### PR TITLE
fix issue that print non-ascii character may cause exception

### PR DESCRIPTION
**Pull request by yushihang**
[UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 20: ordinal not in range(128)]

### CODE REVIEW

File: lldb/logs.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
-
-            print(line.rstrip("\n"), end=None)
+            try:
+                line = file.readline()
+                
+                if not line:
+                    time.sleep(0.1)
+                    continue
+                    
+                print(line.rstrip("\n"), end=None, file=utf8_stdout, flush=True)
```
Comment (nisargjhaveri) [inline lldb/logs.py:19]: Curious: Why is `flush=True` required?

File: lldb/logs.py [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+                
+            except UnicodeDecodeError as e:
+                print(f"UnicodeDecodeError occurred while following file {filename}: f{e}")
```
Comment (nisargjhaveri) [inline lldb/logs.py:22]: Is the `f` in `f{e}` intentional?

Comment (yushihang) [reply lldb/logs.py]: fixed, a mistake while copying , thanks for reviewing

File: lldb/logs.py (hunk 1 above)
Comment (yushihang) [reply lldb/logs.py:19]: https://stackoverflow.com/a/3597849/2201261

I used the advice in this post

**Review by nisargjhaveri — APPROVED** [review]

**Pull request merged**